example/src/App.js

```diff
@@ -10,22 +11,46 @@ export default function App({ initialLocation }) {
 }
 
 function Page() {
-  const { matchRoutes } = useRouting()
-  const routes = routeMap
+  const location = useLocation()
+  const pageRoutePath = usePageRoutePath()
+
+  const transition = useTransition(location, {
+    key: pageRoutePath,
+    from: { opacity: 0, translateX: '-500px' },
+    enter: { opacity: 1, translateX: '0' },
+    leave: { opacity: 0, translateX: '500px' }
+  })
 
   return (
     <>
       <Navigation />
-      {matchRoutes(
-        [routes.home, <Home />],
-        [routes.articles, <Articles />],
-        [routes.articles.article, params => <Article {...{ params }} />],
-        [routes.notFound, params => <NotFound {...{ params }} />],
-      )}
+      {transition((props, transitionLocation) => {
+        const isLeaving = location !== transitionLocation
+        return (
+          <animated.div style={{ ...(isLeaving && { position: 'absolute' }), ...props }}>
+            {isLeaving
+              ? <StaticLocationProvider location={transitionLocation} children={<Content />} />
```

Comment:
Snap ik het goed dat de `StaticLocationProvider` eigenlijk een soort van "page to navigatie to" provider is?

Reply:
Volgens mij niet. Het zorgt ervoor dat de kinderen renderen op basis van een route die misschien niet meer in de url balk staat.